pkg/util/p8s.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"go.uber.org/zap"
+)
+
+var (
+	addToPusher = push.AddFromGatherer
+)
+
+// NewMetricClient returns a pointer to a MetricClient
+func NewMetricClient(addr string, interval time.Duration, registry *prometheus.Registry) *MetricClient {
+	return &MetricClient{addr: addr, interval: interval, registry: registry}
+}
+
+// MetricClient manage the periodic push to the Prometheus Pushgateway.
+type MetricClient struct {
+	addr     string
+	interval time.Duration
+	registry *prometheus.Registry
+}
+
+// Start run a loop of pushing metrics to Prometheus Pushgateway.
+func (mc MetricClient) Start(ctx context.Context, grouping map[string]string) {
```

Comment:
seems all the grouping is `"instance": nodeid`, how about use nodeid as arg

Reply:
@july2993 What's your opinion? Do we want more flexibility here or shorter code?

Reply:
i prefer keep `grouping`